### DISCUSSION TITLE
Additional Packages

### DISCUSSION THREAD

**Discussion by ghost** [itsManjeet/avyos#24] [Ideas]
We can provide addition packages support using our PKGUPD. but the problem is the core system is immutable so we have 3 options (may be more):

1. rpm-ostree like package layering.

I personally don't like the rpm-ostree model of package layering, as it require reboot every time on package additions and I don't know might create a mesh.

2. Overlay over roots 

Need to check the performance overhead, not that good with KDE on rlxos 2016.

3. Seperate ostree inside /opt

We can go with this approach, we can manage a separate repository at /opt/repo and packages at /opt/app1, /opt/app2 and use container --mount /opt/app1:/app --mount /usr:/usr:ro nearly like flatpak,
or
we can export app1, app2 directly to /opt/{bin,etc,share} hierarchy. With this we need to precheck for the duplicates and track the checkouts as ostree has no uncheckout option 😅️.

**Comment by itsManjeet** [comment]
AppImages?
Yes, we need a talk about hosting libraries and CLI.

we can brainstorm on 3rd approach. still, the same issue with how to host libraries and CLI if we go with containers,
The second is good, we can manage independent packages with no duplicates

**Reply by ghost** [reply]
yes, and AppImages are good but can't be used for the distribution of all types of packages, I am doing a POC for 3rd approach and trying some env-like files, that set up environment variables to use that particular package. and can do the same thing with appimages if we ever have to go this way.

**Reply by itsManjeet** [reply]
#26 So we can carry one with the 3rd approach!

**Reply by ghost** [reply]
Layers look good and work familar, like we cheery-pick the rlxos 2107 features in 2307 😄 
how are we going to manage updates and dependencies or it just limited to overlay management?
